pkg/executables/kubectl.go

```diff
@@ -480,26 +480,36 @@ func (k *Kubectl) ValidateControlPlaneNodes(ctx context.Context, cluster *types.
 	return nil
 }
 
-func (k *Kubectl) ValidateWorkerNodes(ctx context.Context, clusterName string, kubeconfig string) error {
+func (k *Kubectl) ValidateWorkerNodes(ctx context.Context, clusterName string, kubeconfig string) (int, int, error) {
 	logger.V(6).Info("waiting for nodes", "cluster", clusterName)
 	deployments, err := k.GetMachineDeployments(ctx, WithKubeconfig(kubeconfig), WithNamespace(constants.EksaSystemNamespace))
 	if err != nil {
-		return err
+		return 0, 0, err
 	}
+	readyReplicas, totalReplicas := CountReplicasByStatus(deployments)
 	for _, machineDeployment := range deployments {
 		if machineDeployment.Status.Phase != "Running" {
-			return fmt.Errorf("machine deployment is in %s phase", machineDeployment.Status.Phase)
+			return readyReplicas, totalReplicas, fmt.Errorf("machine deployment %s is in %s phase", machineDeployment.Name, machineDeployment.Status.Phase)
 		}
 
 		if machineDeployment.Status.UnavailableReplicas != 0 {
-			return fmt.Errorf("%v machine deployment replicas are unavailable", machineDeployment.Status.UnavailableReplicas)
+			return readyReplicas, totalReplicas, fmt.Errorf("%v replicas of machine deployment %s are unavailable", machineDeployment.Status.UnavailableReplicas, machineDeployment.Name)
 		}
 
 		if machineDeployment.Status.ReadyReplicas != machineDeployment.Status.Replicas {
-			return fmt.Errorf("%v machine deployment replicas are not ready", machineDeployment.Status.Replicas-machineDeployment.Status.ReadyReplicas)
+			return readyReplicas, totalReplicas, fmt.Errorf("%v replicas of machine deployment %s replicas are not ready", machineDeployment.Status.Replicas-machineDeployment.Status.ReadyReplicas, machineDeployment.Name)
 		}
 	}
-	return nil
+	return readyReplicas, totalReplicas, nil
+}
+
+func CountReplicasByStatus(deployments []clusterv1.MachineDeployment) (int, int) {
```

Comment:
this doesn't seem to take consideration the same factors that the original method did
for example, it doesn't check `machineDeployment.Status.Phase != "Running"`